PowerShellGet/PSModule.psm1

```diff
@@ -2873,18 +3005,65 @@ function Publish-Script
                                -CallerPSCmdlet $PSCmdlet `
                                -ErrorCategory InvalidOperation
                 }
-                elseif(-not $Force -and ($currentPSGetItemInfo.Version -gt $PSScriptInfo.Version))
+
+                $galleryScriptVer,$galleryScriptPrerelease = $galleryScriptVersion -split '-',2
+                $toPublishScriptVer,$toPublishScriptPrerelease = $toPublishScriptVersion -split '-',2
+
+                if ($galleryScriptVer -eq $toPublishScriptVer)
+                {
+                    # Prerelease strings will not both be null, otherwise would have terminated already above

```

Comment:
Yes, please see above on lines 2993-2996.  I compare the version-prerelease strings before I split them apart.  So if both the versions are the same, and both the prerelease strings are null, then the condition at 2996 would have already been met.  Thus, it is not possible for the prerelease strings to both be null at this point in the code.

---
In reply to: [132591846](https://github.com/rebro-msft/PowerShellGet/pull/1#discussion_r132591846) [](ancestors = 132591846)